fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RelativePriorityPlanSelector.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * RelativePriority.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.plans;
+
+import com.apple.foundationdb.record.ObjectPlanHash;
+import com.apple.foundationdb.record.PlanHashable;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+public class RelativePriorityPlanSelector implements RecordQuerySelectorPlan.PlanSelector {
+    private static final ObjectPlanHash BASE_HASH = new ObjectPlanHash("Relative-Priority-Plan-Selector");
+    @Nonnull
+    List<Double> priorities;
+    @Nonnull
+    Random random;
+
+    public RelativePriorityPlanSelector(@Nonnull final List<Double> priorities) {
+        this(priorities, new Random());
+    }
+
+    @VisibleForTesting
+    RelativePriorityPlanSelector(@Nonnull final List<Double> priorities, @Nonnull final Random random) {
+        Verify.verify(!priorities.isEmpty());
+        this.priorities = ImmutableList.copyOf(priorities);
+        this.random = random;
+        double sumAll = this.priorities.stream().mapToDouble(Double::doubleValue).sum();
+        Verify.verify(sumAll == 1.0D);
```

Comment:
1. use `BigDecimal` -- that may be overkill
2. use rational numbers, i.e. use a base of some number, e.g. 100 and then use dividends thereof
3. or we just change this to `sumAll <= 1.0D` since you are summing up floats that may have been truncated already your sum should only err on the small side me thinks. If you really care about a gap between your sum and `1.0D` you can have special logic pick the last plan if you ever fall into that range.

Reply:
For simplicity, used %-based integer solution. This way, the user will specify a list of int values all adding up to 100.
The reason I went with 1-based doubles before was to allow "unlimited" number of sub-plans that offers better flexibility, but this is really not necessary in this case, where we will have a handful (probably no more than 2 or 3) plans that we would have to choose from, and the idea was to just sanity check that their priorities are reasonable. Since this caused some concerns, we should just opt for a simpler solution.